### PR TITLE
BIGC-739: Fixed tracking and added previous pageTracker logic

### PR DESCRIPTION
**Pull request by Menes1337**
Tracking was updated from logic which was already implemented for the legacy bigcommerce web checkout. 

## Type of change

- [x] Bug fix
- [ ] New feature
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

**Review by sofib — COMMENTED** [review]
I think we should also remove shopgate-analytics and everything related to it as it is no longer used. I could do this but I'm not sure it that is fine with you.

File: src/bigcommerce.js [hunk 1]
```diff
@@ -147,17 +182,52 @@
...
+    maximumIntervalTimeInMilliseconds,
     executeCallback
   ) {
-    const startTimestampInMiliseconds = Date.now();
+    var startTimestampInMilliseconds = Date.now();
 
-    const interval = setInterval(function() {
-      if (startTimestampInMiliseconds + maximumIntervallTimeInMiliseconds <= Date.now()) {
+    var interval = setInterval(function() {
+      if (startTimestampInMilliseconds + maximumIntervalTimeInMilliseconds <= Date.now()) {
```
Comment (chrisEff) [inline src/bigcommerce.js]: I would prefer to have this as
`if (Date.now() >= startTimestampInMilliseconds + maximumIntervalTimeInMilliseconds) {`
which does exactly the same, but is easier to understand for me personally.
(The way it's written atm really confused me. 😄)

**Review by sofib — COMMENTED** [review]
I've tested the code and had issues with SGEvent that didn't appear before. Therefore I've decided to make a hard split between the declaration and code execution as we talked on Thursday.  As I did these changes already I've also removed shopgate-analytics.js and its references in packace.json

I've tested our flows and for me this is ok, but you'll need to check my changes.

**Comment by Menes1337** [comment]
@sofib unfortunate we can't delete shopgate-analytics.js so easily. The file is still needed for merchants who reference this file in the old JavaScript code in BigCommerce in the Google Analytics input field.

**Review by sofib — COMMENTED** [review]
Related to the shopgate-analytics deletion - Integrations with our BigCommerce Connect web checkout has been checked by Nicollette, it is not a problem to update as soon as this is done.

**Review by sofib — APPROVED** [review]

**Pull request merged**